src/main/flight/mixer.c

```diff
@@ -269,83 +274,125 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
+    if (cosPhi < cosThreshold) {
+        // Enforce either roll or pitch exclusively, if not on diagonal
+        if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
             signPitch = 0;
         } else {
-            // If pitch/roll dominant, disable yaw
-            signYaw = 0;
+            signRoll = 0;
         }
+    }
 
-        const float cosPhi = (stickDeflectionLength > 0) ? (stickDeflectionPitchAbs + stickDeflectionRollAbs) / (sqrtf(2.0f) * stickDeflectionLength) : 0;
-        const float cosThreshold = sqrtf(3.0f)/2.0f; // cos(PI/6.0f)
-
-        if (cosPhi < cosThreshold) {
-            // Enforce either roll or pitch exclusively, if not on diagonal
-            if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
-                signPitch = 0;
-            } else {
-                signRoll = 0;
+    // Calculate crashflipPower from stick deflection with a reasonable amount of stick deadband
+    float crashflipPower = stickDeflectionLength > CRASHFLIP_STICK_DEADBAND ? stickDeflectionLength : 0.0f;
+
+    // calculate flipPower attenuators
+    float crashflipRateAttenuator = 1.0f;
+    float crashflipAttitudeAttenuator = 1.0f;
+    const float crashflipRateLimit = mixerConfig()->crashflip_rate * 10.0f; // eg 35 = no power by 350 deg/s
+    const float halfComplete = 0.5f; // attitude or rate changes less that this will be ignored
+
+    // disable both attenuators if the user's crashflip_rate is zero
+    if (crashflipRateLimit > 0) {
+#ifdef USE_ACC
+        // Calculate an attenuator based on change of attitude (requires Acc)
+        // with Acc, crashflipAttitudeAttenuator will be zero after approx 90 degree rotation, and
+        // motors will stop / not spin while attitude remains more than ~90 degrees from initial attitude
+        // without Acc, the user must manually center the stick, or exit crashflip mode, or disarm, to stop the motors
+        // re-initialisation of crashFlip mode by arm/disarm is required to reset the initial tilt angle
+        if (sensors(SENSOR_ACC)) {
+            const float tiltAngle = getCosTiltAngle();  // -1 if flat inverted, 0 when 90° (on edge), +1 when flat and upright
+            if (!isTiltAngleAtStartSet) {
+                tiltAngleAtStart = tiltAngle;
+                isTiltAngleAtStartSet = true;
+                crashflipSuccess = false;
             }
+            // attitudeChangeNeeded is 1.0 at the start, decreasing to 0 when attitude change exceeds approx 90 degrees
+            const float attitudeChangeNeeded = fmaxf(1.0f - fabsf(tiltAngle - tiltAngleAtStart), 0.0f);
+            // no attenuation unless a significant attitude change has occurred
+            crashflipAttitudeAttenuator = attitudeChangeNeeded > halfComplete ? 1.0f : attitudeChangeNeeded / halfComplete;
+
+            // signal success to enable quick restart, if attitude change implies success when reverting the switch
+            crashflipSuccess = crashflipAttitudeAttenuator == 0.0f;
```

Comment:
this may be a  bit easier to understand:
```suggestion
            crashflipSuccess = attitudeChangeNeeded  == 0.0f;
```

Reply:
Yes, it's more transparent like that.